packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/FixedEnumProvider.cs

```diff
@@ -67,10 +69,11 @@ public override ValueExpression ToSerial(ValueExpression enumExpression)
             }
 
             // otherwise we call the corresponding extension method to convert the value
-            return new InvokeStaticMethodExpression(Serialization?.Type, $"ToSerial{ValueType.Name}", [enumExpression], CallAsExtension: true);
+            CSharpType? serializationType = SerializationProviders.FirstOrDefault()?.Type;
+            return new InvokeStaticMethodExpression(serializationType, $"ToSerial{ValueType.Name}", [enumExpression], CallAsExtension: true);
         }
 
         public override ValueExpression ToEnum(ValueExpression valueExpression)
-            => new InvokeStaticMethodExpression(Serialization?.Type, $"To{Type.Name}", [valueExpression], CallAsExtension: true);
+            => new InvokeStaticMethodExpression(SerializationProviders.FirstOrDefault()?.Type, $"To{Type.Name}", [valueExpression], CallAsExtension: true);
```

Comment:
what if they override that method in TypeFactory and returns nothing?
what if they override the enum implementation so that the enum no longer carries a `To{Type.Name}` method?
I do not think the changes in this type makes much sense.

Reply:
Lets open an issue to remove ToEnum and ToSerial from the base EnumProvider class since these are serialization methods and should not be required by the EnumProvider they should be add ons by the SerializationProvider.  This will ensure the name lives with the SerializationProvider and this conflict can't arise.

Reply:
Created https://github.com/microsoft/typespec/issues/3651